lib/stripe/payment_methods/card.ex

```diff
@@ -128,14 +130,21 @@ defmodule Stripe.Card do
   @spec update(source, String.t, String.t, map, Keyword.t) :: {:ok, t} | {:error, Stripe.api_error_struct}
   def update(owner_type, owner_id, card_id, changes, opts \\ []) do
     endpoint = endpoint_for_owner(owner_type, owner_id) <> "/" <> card_id
-    Stripe.Request.update(endpoint, changes, @schema, @nullable_keys, opts)
+    Stripe.Request.update(endpoint, changes, opts)
   end
 
   @doc """
   Delete a card.
   """
-  @spec delete(source, String.t, String.t, Keyword.t) :: :ok | {:error, Stripe.api_error_struct}
-  def delete(owner_type, owner_id, card_id, opts \\ []) do
+  @spec delete(source, owner_or_id, card_or_id, Keyword.t) :: :ok | {:error, Stripe.api_error_struct}
+      when owner_or_id: owner | String.t, card_or_id: t | String.t
+  def delete(owner_type, owner, card, opts \\ []) when owner_type in @sources do
+    owner_id = Util.normalize_id(owner)
+    card_id = Util.normalize_id(card)
+    do_delete(owner_type, owner_id, card_id, opts)
+  end
+
+  defp do_delete(owner_type, owner_id, card_id, opts \\ []) do
```

Comment:
I'm not sure if there was a reason to split this up previously, but you might as well put the two lines in `do_delete` into `delete`, since there's no code reuse happening.